pkg/storage/filesystem/btrfs.go

```diff
@@ -393,18 +416,59 @@ func (p *btrfsPool) Reserved() (uint64, error) {
 	return total, nil
 }
 
+func (p *btrfsPool) Maintenance() error {
+	// this method cleans up all the unused
+	// qgroups that could exists on a filesystem
+
+	volumes, err := p.Volumes()
```

Comment:
Volumes() for both Pool and Subvol actually only lists the first immediate sub volumens. in other words it uses the `btrfs subvol list -o` (not the -o flag) (print only subvolumes below specified path).

Which means if you have sub-volumes of other sub-volumes they won't be cleaned up. Note that while the library supports this, we don't actually expose this via the storage module so there is no "way" that this situation gonna happen.

The question is, should we just do cleaning on the first level of sub-volumes, or make the library code complete by making sure the maintenance do the proper cleaning regardless the depth of the subvol

A quick solution is to implement a SubvolListAll() util and call it directly at this line

Reply:
My full solution assumes we only create a single level of sub volume anyhow. And indeed that's the only thing we support at the moment.
I would be more tempted to remove any code that allow to create sub volume of subvolume and harden the fact that a qgroup is always linked to a single  sub volume, that makes management much easier. What do you say ?

Reply:
Sure we can do that too. I can't think of a scenario where a user would like to create sub-sub-volumes on his volumes.

Reply:
ok, Done in my last commit